packages/frontend/src/components/ControlledAutocomplete/index.tsx

```diff
@@ -76,55 +80,92 @@ function ControlledAutocomplete(
     return rawFreeSolo
   }, [options, rawFreeSolo])
 
-  return (
-    <Controller
-      rules={{ required }}
-      name={name}
-      defaultValue={defaultValue ?? ''}
-      control={control}
-      render={({ field: { ref, onChange, value: fieldValue }, fieldState }) => {
-        const isError = Boolean(fieldState.isTouched && fieldState.error)
+  /**
+   * useController is used here instead of the Controller component
+   * so that we could call the onChange handler
+   */
+  const {
+    field: { value: fieldValue, onChange: fieldOnChange, ref },
+    fieldState: { isTouched, error },
+  } = useController({
+    name,
+    control,
+    rules: { required },
+    defaultValue: defaultValue ?? '',
+  })
+
+  const isError = Boolean(isTouched && error)
 
-        return (
-          <FormControl isInvalid={isError}>
-            {label && (
-              <FormLabel
-                isRequired={required}
-                description={
-                  description && (
-                    <Markdown linkTarget="_blank">{description}</Markdown>
-                  )
-                }
-              >
-                {label}
-              </FormLabel>
-            )}
-            {/* Dropdown row option content */}
-            <Flex>
-              <Box flexGrow={1}>
-                <SingleSelect
-                  name="choose-dropdown-option"
-                  colorScheme="secondary"
-                  isClearable={!required}
-                  items={items}
-                  onChange={onChange}
-                  value={fieldValue}
-                  placeholder={placeholder}
-                  ref={ref}
-                  data-test={`${name}-autocomplete`}
-                  onRefresh={onRefresh}
-                  isRefreshLoading={loading}
-                  freeSolo={freeSolo}
-                />
-              </Box>
-            </Flex>
-            {isError && (
-              <FormErrorMessage>{fieldState.error?.message}</FormErrorMessage>
-            )}
-          </FormControl>
-        )
-      }}
-    />
+  // Control add new option modal
+  const {
+    onClose: onNewOptionModalClose,
+    onOpen: onNewOptionModalOpen,
+    isOpen: isNewOptionModalOpen,
+  } = useDisclosure()
+
+  const { createNewOption, isCreatingNewOption } =
+    useCreateNewOption(fieldOnChange)
+
+  const onNewOptionModalSubmit = useCallback(
+    (inputValue: string) => {
+      onNewOptionModalClose()
+      createNewOption({ inputValue, addNewId: addNewOption?.id })
```

Comment:
thinking if better to close modal only if `createNewOption` succeeds

Reply:
this is to de-dupe the create new option logic for inline and modal type